### PR TITLE
DHFPROD-6051: The connector should point to java-client-api version 5.3

### PR DESCRIPTION
**Pull request by anu3990**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/Options.java [hunk 1]
```diff
@@ -21,8 +21,7 @@
     private String permissions;
     private String sourceName;
     private String sourceType;
-    private JsonNode ingestWorkUnit;
-    private JsonNode ingestEndpointState;
+    private JsonNode ingestEndpointConstants;
```
Comment (rjrudin) [inline marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/Options.java:24]: Don't we still need ingestEndpointState? Or is that not relevant at all to input-only endpoints?

**Review by rjrudin — COMMENTED** [review]
All looks good, just wanted to verify that we don't want to allow Ernie to provide a custom endpointState to his custom endpoint.

File: marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/Options.java (hunk 1 above)
Comment (anu3990) [reply marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/Options.java:24]: Done.

File: marklogic-data-hub-spark-connector/src/test/resources/custom-ingestion-endpoint/endpoint.api [hunk 2]
```diff
@@ -3,13 +3,7 @@
...
     "params": [
         {
-            "name": "endpointState",
-            "datatype": "jsonDocument",
-            "multiple": false,
-            "nullable": true
-        },
-        {
-            "name": "workUnit",
+            "name": "endpointConstants",
```
Comment (rjrudin) [inline marklogic-data-hub-spark-connector/src/test/resources/custom-ingestion-endpoint/endpoint.api:6]: This custom endpoint should still have endpointState in it; that's a critical part of verifying that if Ernie uses a custom endpoint, he can define both endpointState and endpointConstants in it.

File: marklogic-data-hub-spark-connector/src/test/resources/custom-ingestion-endpoint/endpoint.sjs [hunk 3]
```diff
@@ -20,13 +20,11 @@ declareUpdate();
...
 var input;
 
-const state = fn.head(xdmp.fromJSON(endpointState));
-const work = fn.head(xdmp.fromJSON(workUnit));
+const work = fn.head(xdmp.fromJSON(endpointConstants));
 
 const doc = {
   input: input,
-  endpointState: state,
-  workUnit: work
```
Comment (rjrudin) [inline marklogic-data-hub-spark-connector/src/test/resources/custom-ingestion-endpoint/endpoint.sjs:29]: We still want to include this and test for it.

File: marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/WriteDataViaCustomEndpointTest.java [hunk 4]
```diff
@@ -60,8 +60,7 @@ private void thenTheExpectedDocumentExists() {
         assertEquals("apple", doc.get("input").get("fruitName").asText(), message);
         assertEquals("red", doc.get("input").get("fruitColor").asText(), message);
 
-        assertEquals("value", doc.get("workUnit").get("someString").asText(), message);
-        assertEquals(123, doc.get("workUnit").get("someNumber").asInt(), message);
-        assertEquals("hello world", doc.get("endpointState").get("someState").asText(), message);
```
Comment (rjrudin) [inline marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/WriteDataViaCustomEndpointTest.java:65]: We still want to verify that if Ernie has custom endpointState, his endpoint gets access to it.

**Review by rjrudin — CHANGES_REQUESTED** [review]
I'll submit a PR into this - I'm realizing we need some additional tests in WriteDataViaCustomEndpointTest

**Comment by rjrudin** [comment]
Submitted - https://github.com/anu3990/marklogic-data-hub/pull/6/files

**Review by rjrudin — DISMISSED** [review]

**Review by srinathgit — APPROVED** [review]

**Comment by anu3990** [comment]
Included changes in "ingestionService.sjs" and rebased. @rjrudin and @srinathgit need your approvals again.

**Review by rjrudin — APPROVED** [review]

**Review by srinathgit — APPROVED** [review]

**Pull request merged**